File: Assets/Plugins/iOS/Unity.m
//
//  Unity.m
//  Unity-iPhone
//
//  Created by root on 2014/05/11.
//
//

#import "Unity.h"

@interface UnityLocal : NSObject

+ (void)sendMessage:(const char *)type message:(const char *)message;

@end

@implementation Unity

+ (void)call:(const char *)message
{
    [UnityLocal sendMessage:"CallFromIOS" message:message];
}

+ (void)audioio:(const char *)message
{
    [UnityLocal sendMessage:"AudioIOFromOS" message:message];
}

+ (void)d:(const char *)message
{
    [UnityLocal sendMessage:"LogDFromOS" message:message];
}

+ (void)w:(const char *)message
{
    [UnityLocal sendMessage:"LogWFromOS" message:message];
}

@end

@implementation UnityLocal

+ (void)sendMessage:(const char *)type message:(const char *)message
{
	@synchronized (self)
	{
	    UnitySendMessage("Receiver", type, message);
	}
}

@end

File: Projects/Eclipse/jp.co.aquariumy.audioio/src/jp/co/aquariumy/audioio/Unity.java
package jp.co.aquariumy.audioio;

import com.unity3d.player.UnityPlayer;

public class Unity
{

	static public void call(String message)
	{
        Unity.sendMessage(UnitySendMessageType.CALL, message);
	}

	static public void audioio(String message)
	{
        Unity.sendMessage(UnitySendMessageType.AUDIOIO, message);
	}

	static public void d(String message)
	{
        Unity.sendMessage(UnitySendMessageType.LOG_D, message);
	}

	static public void w(String message)
	{
        Unity.sendMessage(UnitySendMessageType.LOG_W, message);
	}

	static private void sendMessage(UnitySendMessageType type, String message)
	{
		UnityPlayer.UnitySendMessage("Receiver", type.getMethod(), message);
	}

}

